From 7949864caa46e46d0e67ea5827628d929ff0418e Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Wed, 22 Oct 2025 15:30:05 -0500
Subject: [PATCH 375/377] atiadlxx: Add stub for
 ADL[2]_Display_SLSMapIndex_Get.

---
 dlls/atiadlxx/atiadlxx.spec   |  4 ++--
 dlls/atiadlxx/atiadlxx_main.c | 27 +++++++++++++++++++++++++--
 2 files changed, 27 insertions(+), 4 deletions(-)

diff --git a/dlls/atiadlxx/atiadlxx.spec b/dlls/atiadlxx/atiadlxx.spec
index 532473b0d5a..0ce97e2f0b9 100644
--- a/dlls/atiadlxx/atiadlxx.spec
+++ b/dlls/atiadlxx/atiadlxx.spec
@@ -376,7 +376,7 @@
 @ stub ADL2_Display_SLSMapConfig_SupportedImageCropType_Get
 @ stub ADL2_Display_SLSMapConfig_Valid
 @ stub ADL2_Display_SLSMapIndexList_Get
-@ stub ADL2_Display_SLSMapIndex_Get
+@ cdecl ADL2_Display_SLSMapIndex_Get(ptr long long ptr ptr)
 @ stub ADL2_Display_SLSMiddleMode_Get
 @ stub ADL2_Display_SLSMiddleMode_Set
 @ stub ADL2_Display_SLSRecords_Get
@@ -936,7 +936,7 @@
 @ stub ADL_Display_SLSMapConfig_Rearrange
 @ stub ADL_Display_SLSMapConfig_SetState
 @ stub ADL_Display_SLSMapIndexList_Get
-@ stub ADL_Display_SLSMapIndex_Get
+@ cdecl ADL_Display_SLSMapIndex_Get(long long ptr ptr)
 @ stub ADL_Display_SLSMiddleMode_Get
 @ stub ADL_Display_SLSMiddleMode_Set
 @ stub ADL_Display_SLSRecords_Get
diff --git a/dlls/atiadlxx/atiadlxx_main.c b/dlls/atiadlxx/atiadlxx_main.c
index 1d840aa5a7b..9236ab12c84 100644
--- a/dlls/atiadlxx/atiadlxx_main.c
+++ b/dlls/atiadlxx/atiadlxx_main.c
@@ -1011,8 +1011,9 @@ int CDECL ADL_Graphics_Platform_Get(int *platform)
     return ADL2_Graphics_Platform_Get(default_ctx, platform);
 }
 
-int CDECL ADL2_Display_DisplayMapConfig_Get(ADL_CONTEXT_HANDLE ctx, int adapter_index, int *display_map_count, ADLDisplayMap **display_maps,
-        int *display_target_count, ADLDisplayTarget **display_targets, int options)
+int CDECL ADL2_Display_DisplayMapConfig_Get(ADL_CONTEXT_HANDLE ctx, int adapter_index, int *display_map_count,
+                                            ADLDisplayMap **display_maps, int *display_target_count,
+                                            ADLDisplayTarget **display_targets, int options)
 {
     struct gpu *gpu;
     int i;
@@ -1075,3 +1076,25 @@ int CDECL ADL_Display_DisplayMapConfig_Get(int adapter_index, int *display_map_c
     return ADL2_Display_DisplayMapConfig_Get(default_ctx, adapter_index, display_map_count, display_maps,
             display_target_count, display_targets, options);
 }
+
+int CDECL ADL2_Display_SLSMapIndex_Get(ADL_CONTEXT_HANDLE ctx, int adapter_index, int num_display_target,
+                                       ADLDisplayTarget *display_target, int *sls_index)
+{
+    FIXME("ctx %p adapter_index %d num_display_target %d display_target %p sls_index %p stub!\n",
+          ctx, adapter_index, num_display_target, display_target, sls_index);
+
+    if (!ctx || !display_target || !sls_index) return ADL_ERR_INVALID_PARAM;
+
+    *sls_index = 0;
+
+    return ADL_OK;
+}
+
+int CDECL ADL_Display_SLSMapIndex_Get(int adapter_index, int num_display_target,
+                                      ADLDisplayTarget *display_target, int *sls_index)
+{
+    TRACE("adapter_index %d num_display_target %d display_target %p sls_index %p\n",
+          adapter_index, num_display_target, display_target, sls_index);
+
+    return ADL2_Display_SLSMapIndex_Get(default_ctx, adapter_index, num_display_target, display_target, sls_index);
+}
-- 
2.51.0

